swap-install/src/lib.rs

```diff
@@ -0,0 +1,39 @@
+#![cfg_attr(not(test), no_std)]
+
+extern crate alloc;
+
+use alloc::{collections::BTreeMap, string::String};
+
+use contract::contract_api::{runtime, storage, system};
+use types::{Key, URef};
+
+const KEY_ADMIN: &str = "admin";
+const NAME_SWAP_HASH: &str = "swap_hash";
+const NAME_SWAP_LOGIC_EXT: &str = "swap_logic_ext";
+const KEY_CONTRACT_PURSE: &str = "swap_contract_purse";
+
+#[no_mangle]
+pub extern "C" fn swap_logic_ext() {
+    swap_logic::delegate();
+}
+
+#[no_mangle]
+pub extern "C" fn call() {
+    // create map of references for stored contract
+    let mut swapper_urefs: BTreeMap<String, Key> = BTreeMap::new();
+
+    // Get caller's public key and store as admin
+    let admin_uref: URef = storage::new_uref(runtime::get_caller());
+    swapper_urefs.insert(String::from(KEY_ADMIN), admin_uref.into());
+
+    // Create purse for contract
+    let contract_purse = system::create_purse();
+    let contract_purse_uref = storage::new_uref(contract_purse);
```

Comment:
It seems okay to store the `contract_purse` directly to swapper_urefs.
Is there a reason to store the purse again under another uref?

Reply:
Don't want to expose Purse URef externally by querying

Reply:
https://github.com/hdac-io/swap/pull/1/commits/2115165493de5345b08994e3bf1e68853a65870f